Feliz.CompilerPlugins/ReactComponent.fs

```diff
@@ -0,0 +1,71 @@
+﻿namespace Feliz
+
+open Fable
+open Fable.AST
+
+/// <summary>Transforms a function into a React function component. Make sure the function is defined at the module level</summary>
+type ReactComponentAttribute() =
+    inherit MemberDeclarationPluginAttribute()
+    override _.FableMinimumVersion = "3.0"
+
+    /// <summary>Transforms call-site into createElement calls</summary>
+    override _.TransformCall(logger, memb, expr) =
+        let membArgs = memb.CurriedParameterGroups |> List.concat
+        match expr with
+        | Fable.Call(callee, info, typeInfo, range) when List.length membArgs = List.length info.Args ->
+            let callee =
+                match callee with
+                | Fable.Expr.IdentExpr ident -> Fable.Expr.IdentExpr { ident with Name = AstUtils.capitalize ident.Name }
+                | _ -> callee
```

Comment:
If we transform the declaration name, here we should also check if this is an import (when the component is in another file) and capitalize the `Selector` property. Although I would personally prefer if user just capitalize components by themselves as a convention (to distinguish them from just "helper" functions) 😄

Reply:
> Although I would personally prefer if user just capitalize components by themselves as a convention (to distinguish them from just "helper" functions) 😄

Chances are that this is rather easy to forget and goes against F# style guidelines to define functions within a module in camelCase format, so if it is relatively easy to do then why not :smile:

Reply:
@alfonsogarciacaro This might be harder than I thought it would be 😓 I can only transform the call-site but I cannot transform the generated import in the file:
```js
import { counter as Examples_counter } from "./Examples.fs.js" 
```
and inside `Example.fs.js`, the function is exported as PascalCase because of the transformation
```js
export function Counter() { 
 // ... 
}
```
Shouldn't the import be using the transformed `declaration.Name` instead of the default generated one by Fable?

Reply:
Hmm, not sure I understand. If you change the `Selector` property in the Import expression, the generated JS import should change as well. Did you try something like this?

```fsharp
| Fable.IdentExpr ident -> Fable.IdentExpr { ident with Name = capitalize ident.Name }
| Fable.Import(info, t, r) -> Fable.Import({ info with Selector = capitalize info.Selector }, t, r)
| _ -> callee
```